src/main/java/com/mongodb/jdbc/MongoDatabaseMetaData.java

```diff
@@ -116,8 +137,8 @@ public static String escapeString(String value) {
         return escaped.replace("\\", "\\\\");
     }
 
-    public static String toJavaPattern(String sqlPattern) {
-        return sqlPattern.replaceAll("%", ".*").replaceAll("_", ".");
+    public static Pattern toJavaPattern(String sqlPattern) {
+        return Pattern.compile(sqlPattern.replaceAll("%", ".*").replaceAll("_", "."));
```

Comment:
every call to this method looked the same:
```
Pattern xyz = Pattern.compile(toJavaPattern(sqlPattern));
```
so I decided to just do the `Pattern` creation _in_ the method itself.